Assets/js/03.campaigns.js

```diff
@@ -68,6 +68,15 @@ Mautic.contactsourceCampaigns = function () {
                         campaignsJSONEditor.on('change', function (event) {
                             var obj = campaignsJSONEditor.getValue();
                             if (typeof obj === 'object') {
+                                var campaignIds = [];
+                                obj.campaigns.forEach(function(cmpn) { 
+                                    if(campaignIds.indexOf(cmpn.campaignId) >= 0) {
+                                        alert("You shouldn't add the same campaign to a Source more than once, create a different campaign instead.");
+                                    }
+                                    campaignIds.push(cmpn.campaignId);
+                                });
+
+
```

Comment:
I'm afraid this isn't enough. Our users may just skip this alert... Any way we can block the addition instead?

Reply:
It now reverts the selected campaign to zero.